chia/types/weight_proof.py

```diff
@@ -100,3 +103,63 @@ class WeightProof(Streamable):
     sub_epochs: List[SubEpochData]
     sub_epoch_segments: List[SubEpochChallengeSegment]  # sampled sub epoch
     recent_chain_data: List[HeaderBlock]
+
+
+@streamable
+@dataclass(frozen=True)
+class SubSlotDataV2(Streamable):
+    proof_of_space: Optional[ProofOfSpace]  # if challenge block
+    # VDF to signage point
+    cc_signage_point: Optional[VDFProof]  # if block
+    # VDF from signage to infusion point
+    cc_infusion_point: Optional[VDFProof]  # if block
+    signage_point_index: Optional[uint8]  # if block
+    cc_slot_end: Optional[VDFProof]  # if is end of slot
+    cc_sp_vdf_output: Optional[CompressedClassgroupElement]  # if is end of slot
+    cc_ip_vdf_output: Optional[CompressedClassgroupElement]  # if is end of slot
+    cc_slot_end_output: Optional[ClassgroupElement]  # if is end of slot
```

Comment:
Why not compressed?

Reply:
getting from the compressed form (B) to the regular ClassgroupElement means we need to validate the vdf (its almost the same code)

that means that if we use the B form for the slots we cant really validate the hashes without validating the slot vdfs, and that means we cant really use the optimization where we validate vdfs only for one challenge slot in each sub_epoch